resources/resource_transformers/js/js.go

```diff
@@ -55,17 +55,17 @@ func New(fs *filesystems.SourceFilesystem, rs *resources.Spec) *Client {
 	return &Client{rs: rs, sfs: fs}
 }
 
-type esbuildTransformation struct {
+type jsTransformation struct {
 	options Options
 	rs      *resources.Spec
 	sfs     *filesystems.SourceFilesystem
 }
 
-func (t *esbuildTransformation) Key() internal.ResourceTransformationKey {
-	return internal.NewResourceTransformationKey("esbuild", t.options)
+func (t *jsTransformation) Key() internal.ResourceTransformationKey {
```

Comment:
All of this looks good, I have one final "suggestion":

As we will (most likely) get more transformations in this package, I suggest that you name this file `build.go`, the key `jsbuild` and the struct `buildTransformation`. That will leave room for `jstransform` etc.

Reply:
Also, this need a test. The simplest is to just expand on the tests in /hugolib.

Reply:
@bep Is this enough as a test? This tests the filesystem interaction.

If you want me to test various options, I'll have to create a test at a lower level (i don't need the filesystem for those)